### PR TITLE
Don't depend on mingwpy where it doesn't exist

### PR DESCRIPTION
**Pull request by inducer**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ requirements:
         - numpy
         - ocl-icd  # [linux]
         - khronos-opencl-icd-loader  # [win]
-        - mingwpy  # [win]
+        - mingwpy  # [win and (py27 or py34 or py35)]
```
Comment (isuruf) [inline recipe/meta.yaml]: Setting this to `# [win and py!=36]` works.

Comment (inducer) [inline recipe/meta.yaml]: How about `py<=35`?

Comment (isuruf) [inline recipe/meta.yaml]: Tried a number of ways including that. Only `py!=36` works

Comment (inducer) [inline recipe/meta.yaml]: Thanks for checking! I'm not too fond of that because it'll promptly break at `py37`.

**Comment by inducer** [comment]
@isuruf, on a semi-related note: Do you have any suggestions on how to coax this into building for Python 3.6 on Windows? That was the objective of the exercise, but it ostensibly (see the builds on Appveyor) didn't work.

**Comment by isuruf** [comment]
Hmm, you mean that rerendering doesn't add the py36 builds (Using `py!=36` adds it) or that the build fails without `mingwpy`?

**Comment by inducer** [comment]
No, I had misunderstood--I thought you were merely pointing out an equivalent simpler spelling. I didn't understand that your change fixed the 3.6 builds. With your hint, I've managed to have it enable the 3.6 in a way that won't break at 3.7--using the fact that 3.4 is no longer a thing. :)

Thanks for your help!

**Comment by inducer** [comment]
@isuruf, the 3.6 Windows build failed with a pretty cryptic message. Any hints?

**Comment by isuruf** [comment]
You are using an older image with no `Miniconda36`
```
os:
   - Visual Studio 2015 Update 2
```

Default image now has update 3, so you can just drop this.

**Comment by inducer** [comment]
Heh. That's there on purpose because Update 3 [won't build PyOpenCL](https://github.com/conda-forge/pyopencl-feedstock/pull/1#issuecomment-258619268).

File: appveyor.yml [hunk 2]
```diff
@@ -32,6 +32,16 @@ environment:
...
 
+    - TARGET_ARCH: x86
+      OSX_VARIANT: native
+      CONDA_PY: 36
+      CONDA_INSTALL_LOCN: C:\\Miniconda36
+
+    - TARGET_ARCH: x64
+      OSX_VARIANT: native
+      CONDA_PY: 36
+      CONDA_INSTALL_LOCN: C:\\Miniconda36-x64
```
Comment (isuruf) [inline appveyor.yml]: Change this to `C:\\Miniconda35-x64`

File: appveyor.yml [hunk 3]
```diff
@@ -32,6 +32,16 @@ environment:
       CONDA_PY: 35
       CONDA_INSTALL_LOCN: C:\\Miniconda35-x64
 
+    - TARGET_ARCH: x86
+      OSX_VARIANT: native
+      CONDA_PY: 36
+      CONDA_INSTALL_LOCN: C:\\Miniconda36
```
Comment (isuruf) [inline appveyor.yml]: Change this to `C:\\Miniconda35`

File: appveyor.yml (hunk 2 above)
Comment (inducer) [inline appveyor.yml]: Done, thanks for the suggestion!

**Pull request merged**